crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -199,6 +218,47 @@ pub struct QuerySetState<T>(T);
 
 impl_query_set!();
 
+/// A set of possibly conflicting [`SystemParam`]s which can be accessed one at a time.
+///
+/// The type parameter of a [`ParamSet`] is a tuple of up to 4 [`SystemParam`]s
```

Comment:
```suggestion
/// This is useful when you need to access the same data in different ways within a single system.
/// For simple bundling of system parameters, see the [`SystemParam`] derive instead.
///
/// The type parameter of a [`ParamSet`] is a tuple of up to 4 [`SystemParam`]s
```

More context on why you might want to use this.